scripts/vagrant_lock.sh

```diff
@@ -0,0 +1,65 @@
+# Copyright 2016 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+##########################################################################
+
+# This file should not be invoked directly, but sourced from other sh scripts.
+
+# Creates a lockfile to help with new user confusion when launching a vagrant
+# vm. See https://github.com/oppia/oppia/pull/2749 for details.
+#
+# It can be overridden by passing --nolock to start.sh
+
+
+# This keeps all the arguments passed in, except for --nolock just in case
+# its unexpected presence makes something downstream upset
+declare -a keep_args
+for arg in "$@"; do
+  case $arg in
+    --nolock)
+      NO_LOCK=true
+      ;;
+    *)
+    keep_args+=($arg)
+    ;;
+  esac
+done
+
+set -- "${keep_args[@]}"
+
+[ $NO_LOCK ] && return 0
```

Comment:
Prefer using an ``if`` block since it's a bit more readable, and tends to be waht we do in these situations.